crates/ethereum/payload/src/lib.rs

```diff
@@ -45,6 +46,9 @@ use revm_primitives::calc_excess_blob_gas;
 use std::sync::Arc;
 use tracing::{debug, trace, warn};
 
+type BestTransactionsIter<Transaction> =
+    Box<dyn BestTransactions<Item = Arc<ValidPoolTransaction<Transaction>>>>;
```

Comment:
let's make this generic over pool

`type BestTransactionsFor<Pool> = Box<dyn BestTransactions<Item = Arc<ValidPoolTransaction<<Pool as TransactionPool>::Transaction>>>>`